extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/output/AzureStorageConnector.java

```diff
@@ -73,7 +77,7 @@ public ChunkingStorageConnectorParameters<AzureInputRange> buildInputParams(Stri
   public ChunkingStorageConnectorParameters<AzureInputRange> buildInputParams(String path, long from, long size)
   {
     ChunkingStorageConnectorParameters.Builder<AzureInputRange> parameters = new ChunkingStorageConnectorParameters.Builder<>();
-    parameters.tempDirSupplier(config::getTempDir);
+    parameters.tempDirSupplier(() -> tempFile);
```

Comment:
There doesn't seem to be much benefit to keeping it a supplier. We should refactor Supplier<File> to File

Reply:
Refactored this